Sources/PackageGraph/PackageModel+Extensions.swift

```diff
@@ -23,30 +23,29 @@ extension PackageDependencyDescription {
         //        We should instead use the declared URL of a package dependency
         //        as its identity, as it will be needed for supporting package
         //        registries.
-        let identity = PackageReference.computeIdentity(packageURL: effectiveURL)
-        
         return PackageReference(
```

Comment:
should PackageReference be changed to carry the identify, or this is beyond the scope of this PR?

cc @neonichu

Reply:
My goal for this PR was to swap out the implementation without changing the interface. Once we've done that and are satisfied that we haven't broken anything, my next step was going to be changing `PackageReference` to better fit the new semantics. My working plan is to make `PackageReference` an enumeration (https://github.com/apple/swift-package-manager/pull/3023/commits/095836a247335a89df054de8e5c758dd7af8acc1), with its identity becoming a derived / computed property.

Reply:
my concern in the two step approach is ending up with a mixed bag, but lets see what @neonichu thinks.

Reply:
See https://github.com/apple/swift-package-manager/pull/3032#discussion_r521326243